proto/wechaty/puppet/friendship.proto

```diff
@@ -60,10 +60,18 @@ message FriendshipSearchWeixinResponse {
   google.protobuf.StringValue contact_id = 1;
 }
 
+
+message FriendshipAddOptions {
+  string hello = 1;
+  string room_id = 2;
+  string contact_id = 3;
+}
 message FriendshipAddRequest {
   string contact_id = 1;
   string hello = 2;
+  FriendshipAddOptions options = 3;
```

Comment:
1. I believe we also need an `optional` here because they are all optional.
2. We can put `room_id` and `contact_di` into the `FriendshipAddRequest` so that we will no need to create a new `FriendshipAddOptions`.

Reply:
> 2\. We can put `room_id` and `contact_di` into the `FriendshipAddRequest` so that we will no need to create a new `FriendshipAddOptions`.

how to deal with the double `contact_id` in `FriendshipAddRequest` if no `FriendshipAddOptions`?

Reply:
I would vote for putting these arguments into the options message, instead of flat the object. If we flat the `FriendshipAddRequest`, there will be a inconsistency between the grpc method and puppet methods.

So basically I vote for keeping the grpc methods consistent with puppet methods.